api/src/middleware/merge-content-versions.ts

```diff
@@ -0,0 +1,37 @@
+import type { RequestHandler } from 'express';
+import { VersionsService } from '../services/versions.js';
+import asyncHandler from '../utils/async-handler.js';
+import { mergeVersionsRaw, mergeVersionsRecursive } from '../utils/merge-version-data.js';
+
+export const mergeContentVersions: RequestHandler = asyncHandler(async (req, res, next) => {
+	if (
+		req.sanitizedQuery.version &&
+		req.collection &&
+		(req.singleton || req.params['pk']) &&
+		'data' in res.locals['payload']
+	) {
+		const versionsService = new VersionsService({ accountability: req.accountability ?? null, schema: req.schema });
+
+		const versionData = await versionsService.getVersionSaves(
+			req.sanitizedQuery.version,
+			req.collection,
+			req.params['pk'],
+		);
+
+		if (!versionData) return next();
+
+		const originalData = res.locals['payload'].data;
+
+		if ((originalData && typeof originalData !== 'object') || versionData.length === 0) {
```

Comment:
We can move the `originalData` check to before L13 or as an additional check in the `if`  to reduce work needed for the false case.

I believe we can also reduce the check from `originalData && typeof originalData !== 'object'` to `typeof originalData !== 'object'`? I believe they are equivalent.

Reply:
Right, always with the gotchas. I was thinking more undefined than null, I have not checked the response types yet. I guess we will need to stick with it for now in that case.

Reply:
I feel like this line could need a comment - what the check is supposed to do